### PR TITLE
Adapt to new PyTRiP, inital field dialog work

### PR DESCRIPTION
**Pull request by nbassler**
This MR:
- fixes compability with recent PyTRiP98
- add more documentation 
- update README.md
- adds initial work for the field dialog

### CODE REVIEW

File: pytripgui/controller/plan_cont.py [hunk 1]
```diff
@@ -39,7 +42,11 @@ def new_plan(model):
 
         # open a dialog for the user to edit it
         PlanController.edit_plan(model, plan)
+
+        logger.debug("model.plans.append(plan)")
         model.plans.append(plan)
+        logger.debug(plan)
+        # TODO: TreeController.update_tree()
```
Comment (nbassler) [inline pytripgui/controller/plan_cont.py]: I need some suggestions here, what is the best way to comminuate to TreeController?

File: pytripgui/controller/main_cont.py [hunk 2]
```diff
@@ -197,9 +197,8 @@ def open_voxelplan(self, ctx_path):
         # Check if there is a VDX file with the same basename
         logger.debug("Check for VDX")
         from pytrip.util import TRiP98FilePath
-        _b = TRiP98FilePath(ctx_path, ctx).basename
-        _n = TRiP98FilePath(ctx_path, ctx).name
-        vdx_path = ctx_path.replace(_n, _b) + '.vdx'
+        _d = TRiP98FilePath(ctx_path, ctx).dir_basename  # returns full path, but without suffix.
```
Comment (nbassler) [inline pytripgui/controller/main_cont.py]: new behaviour in pytrip.

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,15 @@
+# generated UIs
```
Comment (grzanka) [inline .gitignore]: What is the problem in adding generated UIs here ?

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,15 @@
...
+# pytripgui/view/gen/field.py
+# pytripgui/view/gen/kernel.py
+# pytripgui/view/gen/main_window.py
+# pytripgui/view/gen/plan.py
+# pytripgui/view/gen/trip_config.py
+
+# versioneer stuff  # TODO find some solution
```
Comment (grzanka) [inline .gitignore]: versioneer cannot be exclude as it needs to add himself temporarily to versioneed files to figureout the version number.
The idea would be to get rid of versioneer at all

**Comment by nbassler** [comment]
Travis complains about missing files then.

File: .gitignore (hunk 3 above)
Comment (nbassler) [reply .gitignore]: Travis complains about missing files then.

File: .gitignore (hunk 4 above)
Comment (nbassler) [reply .gitignore]: Until then, a hack could be to add some of the generated files to the repo? The contents of those files does not seem to change anyway, so maybe it is acceptable.

**Comment by ljelen** [comment]
I've merge our branches @nbassler. Could you take a look at the code now, and test it?

File: pytripgui/controller/field_cont.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
+import logging
+
+from PyQt5 import QtWidgets
+# from PyQt5.QtCore import Qt
```
Comment (grzanka) [inline pytripgui/controller/field_cont.py]: @ljelen if not needed - please remove

File: pytripgui/controller/main_cont.py (hunk 2 above)
Comment (grzanka) [reply pytripgui/controller/main_cont.py]: looks OK :)

**Pull request merged**